### DISCUSSION TITLE
添加对 Thunderbird 的支持 / Add support for Thunderbird

### DISCUSSION THREAD

**Discussion by ShellWen** [immersive-translate/immersive-translate#1696] [Ideas]
### 新功能 | New Feature

添加对 Thunderbird 的支持，开发针对 Thunderbird 的插件，允许沉浸式翻译邮件标题和正文。

Add support for Thunderbird and develop a plugin for Thunderbird that allows immersive translation of email titles and body.

### 为什么您想要添加这些功能 | Why you want to add these features

Thunderbird 是一个广泛使用的邮件客户端，开发针对 Thunderbird 的版本，可以进一步扩展插件的用户覆盖群体。

Thunderbird is a widely used email client, and developing a version for Thunderbird would further extend the plugin's user reach.

### 提议 | Proposal

Thunderbird 虽然是一个邮件客户端，但它的插件开发应该与 Firefox 浏览器无异，可能需要一些针对邮件正文等的特殊的适配。

Thunderbird is an email client, but its plugin development should be similar to that of the Firefox browser, and may require some special works.

**Comment by Vigilans** [comment]
Thunderbird 与 Firefox 使用同样的插件框架进行开发，只需要对一些 Thunderbird 的独有元素进行处理即可，对沉浸式翻译插件来说应该只需要关心邮件页面，也就是 `about:message` 部分即可。

关于最新版 Thunderbird 的 Mail Frontend 的参考文档：
* https://developer.thunderbird.net/thunderbird-development/codebase-overview/mail-front-end
* https://developer.thunderbird.net/add-ons/updating/tb115/adapt-to-changes-in-thunderbird-103-115

一些可供参考的十分简易地对`about:message`页进行处理的插件：
* https://github.com/dillenger/compact-headers

**Comment by TaoLoading** [comment]
你好，本仓库 Releases 中已上线 Thunderbird 版插件安装包（1.19.4）

**Reply by ShellWen** [reply]
您好，[1.19.5](https://github.com/immersive-translate/immersive-translate/releases/tag/v1.19.5) 是不是遗漏了 Thunderbird

**Reply by TaoLoading** [reply]
已更新～

**Comment by nbzz** [comment]
您好，能否再更新一下Thunderbird的插件版本，另外插件的名字太长了，会导致沉浸式翻译一个插件占据thunderbird顶部空间一半的尺寸。
<img width="2932" height="78" alt="image" src="https://github.com/user-attachments/assets/38af1789-8f40-4780-852b-bd67c870d7d2" />
